samples/CMakeLists.txt

```diff
@@ -15,5 +15,7 @@ install(DIRECTORY helloworld file-encryptor data-sealing
 install(FILES README.md
   DESTINATION ${CMAKE_INSTALL_DATADIR}/openenclave/samples)
 
-add_test(NAME samples
-  COMMAND ${CMAKE_COMMAND} -DUSE_LIBSGX=${USE_LIBSGX} -DSOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR} -DBUILD_DIR=${PROJECT_BINARY_DIR} -DPREFIX_DIR=${CMAKE_INSTALL_PREFIX} -P ${CMAKE_CURRENT_SOURCE_DIR}/test-samples.cmake)
+if (NOT OE_PREFIX_MBEDTLS_SYMBOLS)
```

Comment:
Why does this break? Can we fix it?

Reply:
Some of the samples use mbedtls directly, so those won't work. If we have each sample as a different test, then we can fix it. I'd like this in general, because it would allow testing the installation of the `OE_PREFIX_MBEDTLS_SYMBOLS` path without manually running the samples.

Reply:
Can they be updated to use oecrypto instead of mbedtls? I don't recall them using mbedtls directly (at least from a build standpoint), they just picked up whatever was necessary from `oeenclave`/`oelibc` and `oehost`.

The samples test is an integration test of the package; they're not each a different test because the context switches from our build to the installed build when that script is called. They're actually built and ran from an installed copy of the SDK (`build/install/opt/openenclave/...`) (which was installed on the fly by that script). I don't think it's viable to make each sample build and run as a separate test in our build because then each sample would have to install the SDK.

It would require a lot of hacking up of the script. Is there another way?

Reply:
oecrypto is a private API and not available for the installation. The samples use mbedtls directly, since our default OE SDK installation packages mbedtls libraries + headers, so they can call it. (e.g. https://github.com/Microsoft/openenclave/blob/master/samples/data-sealing/common/keys.cpp). 

The idea for `OE_PREFIX_MBEDTLS_SYMBOLS` is that you don't want mbedtls packaged with OE SDK, because your application is using it's own copy of mbedtls which would conflict with the OE one. Since the OESDK still needs crypto, we rename the mbedtls symbols to avoid the symbol issues.

Can we export the  `OE_PREFIX_MBEDTLS_SYMBOLS` for the samples? or maybe some hack that checks if mbedtls is available and only run some of the samples if it is?

Reply:
> Can we export the OE_PREFIX_MBEDTLS_SYMBOLS for the samples?

Can you expand on this?

>  or maybe some hack that checks if mbedtls is available and only run some of the samples if it is?

That's certainly doable, and that might even be something we want to make not a hack at all and instead expose as a a user visible value in the CMake package so that if you're using a build of the SDK without mbedtls, it's very obvious (and easy to test for).

Reply:
> Can you expand on this?

Like in `samples/CMakeLists.txt`: the `add_test` line can have:

```${CMAKE_COMMAND} -DOE_PREFIX_MBEDTLS_SYMBOLS =${OE_PREFIX_MBEDTLS_SYMBOLS } ...```.

Then, use the variable in `test-samples.cmake`.

> That's certainly doable, and that might even be something we want to make not a hack at all and instead expose as a a user visible value in the CMake package so that if you're using a build of the SDK without mbedtls, it's very obvious (and easy to test for).

That sounds reasonable. Is it possible to add a CMake variable to the `openenclave-targets` export set?

Reply:
That'd the the hacky way to do it just for the samples. We don't need do that.

If this is something that users might need to check when they're using the SDK (which it seems like it is), let's define it on package creation in [openenclave-config.cmake.in](https://github.com/Microsoft/openenclave/blob/master/cmake/openenclave-config.cmake.in) instead and use it directly.

> That sounds reasonable. Is it possible to add a CMake variable to the openenclave-targets export set?

 I mean kinda, just add it to that file like the other variables. They'll be filled in on configuration when the file is [configured](https://github.com/Microsoft/openenclave/blob/a6eea72472a798798d1001fc80787b1b92d811c1/cmake/package_settings.cmake#L31).

Then the samples (which use our SDK via `find_package`) will have any and all variables from that file set.